cmd/global-heal.go

```diff
@@ -98,19 +98,42 @@ func healErasureSet(ctx context.Context, setIndex int, xlObj *xlObjects) error {
 	// Heal all buckets with all objects
 	for _, bucket := range buckets {
 		// Heal current bucket
-		bgSeq.sourceCh <- bucket.Name
+		bgSeq.sourceCh <- healSource{
+			path: bucket.Name,
+		}
 
 		// List all objects in the current bucket and heal them
 		listDir := listDirFactory(ctx, xlObj.getLoadBalancedDisks()...)
 		walkResultCh := startTreeWalk(ctx, bucket.Name, "", "", true, listDir, nil)
 		for walkEntry := range walkResultCh {
-			bgSeq.sourceCh <- pathJoin(bucket.Name, walkEntry.entry)
+			bgSeq.sourceCh <- healSource{
+				path: pathJoin(bucket.Name, walkEntry.entry),
+			}
 		}
 	}
 
 	return nil
 }
 
+// deepHealObject heals given object path in deep to fix bitrot.
+func deepHealObject(ctx context.Context, objectPath string) {
+	// Get background heal sequence to send elements to heal
+	var bgSeq *healSequence
+	var ok bool
+	for {
+		bgSeq, ok = globalBackgroundHealState.getHealSequenceByToken(bgHealingUUID)
+		if ok {
+			break
+		}
+		time.Sleep(time.Second)
```

Comment:
This shouldn't block if it cannot queue the heal. Just return (and remove the for loop).

Maybe log an error, since I assume this case isn't expected.

Reply:
I just adapted same logic used in existing code. Not sure why its written this way. @vadmeste could you share your thoughts?

Reply:
> @vadmeste could you share your thoughts?

Yeah the sleep and retry is not needed.

Reply:
@balamurugana The main difference is that this is called from active requests which we don't want to hold up.

Reply:
Removed `for` loop and `time.Sleep()`.